source/scripts/ContentScript/stargazerRequestsProxy.ts

```diff
@@ -0,0 +1,43 @@
+import { browser, Runtime } from 'webextension-polyfill-ts';
+import {
+  StargazerEncodedProxyRequest,
+  StargazerProxyRequest,
+  StargazerEncodedProxyResponse,
+  StargazerProxyResponse,
+} from '../common/proxy-types';
+
+const isStargazerProxyRequest = (value: any, once: string): value is StargazerEncodedProxyRequest => {
+  return 'once' in value && value.once === once;
+};
+
+class StargazerRequestsProxy {
+  #once: string;
+  #port: Runtime.Port;
+
+  constructor(once: string) {
+    this.#once = once;
+    this.#port = browser.runtime.connect(undefined, { name: 'stargazer' });
+  }
+
+  listen() {
+    window.addEventListener('message', this.onWindowMessage.bind(this));
+    this.#port.onMessage.addListener(this.onPortMessage.bind(this));
+  }
+
+  onWindowMessage(event: MessageEvent<unknown>) {
+    if (event.source !== window) {
+      // NOOP
+      return;
+    }
+
+    const encodedRequest = event.data;
+    if (!isStargazerProxyRequest(encodedRequest, this.#once)) {
+      // NOOP
+      return;
+    }
+  }
```

Comment:
Don't we need to do something here if it is a proxy request?

Reply:
Still WIP